File: jyChoiReport/Scens/DetailNews/DetailNewsViewController.m
//
//  DetailNewsViewController.m
//  jyChoiReport
//
//  Created by JuYoung choi on 4/21/24.
//

#import "DetailNewsViewController.h"
#import "jyChoiReport-Swift.h"

@interface DetailNewsViewController ()

@end

@implementation DetailNewsViewController

- (id) init:(NSString *) title url:(NSString *) url {
    
    self = [super initWithNibName:nil bundle:nil];
    
    self.url = url;
    self.newsTitle = title;
    
    return self;
}

- (void) startWebViewIndigator {
    UIActivityIndicatorView *indigator = [[UIActivityIndicatorView alloc] initWithActivityIndicatorStyle:UIActivityIndicatorViewStyleLarge];
    
    indigator.tintColor = [UIColor darkGrayColor];
    
    [self.view addSubview: indigator];
    
    [indigator startAnimating];
    
    indigator.center = self.view.center;
}

- (void) stopWebViewIndigator {
    
    for(int i = 0; i<self.view.subviews.count; i++) {
        
        if ([self.view.subviews[i] isKindOfClass: UIActivityIndicatorView.class ]) {
            
            UIActivityIndicatorView * indigator = ((UIActivityIndicatorView *)self.view.subviews[i]);
            [indigator stopAnimating];
            [indigator removeFromSuperview];
            break;
        }
    }
}

- (void) uiSetting {
    
    CustomWKWebView *webView = [[CustomWKWebView alloc] init];
   
    webView.translatesAutoresizingMaskIntoConstraints = FALSE;
    [self.view addSubview:webView];
    self.webView = webView;
    
    [webView setWebviewDelegate: self];
    [webView requestUrl: self.url];
    [webView setUseProgressBar];
    
    NSLayoutConstraint *leading = [NSLayoutConstraint constraintWithItem:webView attribute:NSLayoutAttributeLeading relatedBy:NSLayoutRelationEqual toItem:self.view attribute:NSLayoutAttributeLeading multiplier:1.0f constant:0.0];
    
    NSLayoutConstraint *trailing = [NSLayoutConstraint constraintWithItem:webView attribute:NSLayoutAttributeTrailing relatedBy:NSLayoutRelationEqual toItem:self.view attribute:NSLayoutAttributeTrailing multiplier:1.0f constant:0.0];
    
    NSLayoutConstraint *top = [NSLayoutConstraint constraintWithItem:webView attribute:NSLayoutAttributeTop relatedBy:NSLayoutRelationEqual toItem:self.view attribute:NSLayoutAttributeTop multiplier:1.0f constant:0.0];
    
    NSLayoutConstraint *bottom = [NSLayoutConstraint constraintWithItem:webView attribute:NSLayoutAttributeBottom relatedBy:NSLayoutRelationEqual toItem:self.view attribute:NSLayoutAttributeBottom multiplier:1.0f constant:0.0];
    
    [self.view addConstraints:@[leading, trailing, top, bottom]];
}

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    
    [self uiSetting];
    self.title = self.newsTitle;
    self.navigationController.navigationBar.tintColor = UIColor.grayColor;
}

// MARK: - CustomWebview delegate
- (BOOL)isHttpRequestUrlWithCallUrl:(NSURL * _Nullable)callUrl {
    return TRUE;
}

- (void) loadingFinish {
    
    if ([self.webView.title isEqual: @""]) {
        
        return;
    }
    
    self.title = self.webView.title;
}

@end

File: jyChoiReport/Scens/DetailNews/_DetailNewsViewController.swift
//
//  DetailNewsViewController.swift
//  jyChoiReport
//
//  Created by JuYoung choi on 4/21/24.
//

import UIKit
import SnapKit

/// 뉴스 상세
class DetailNewsViewController: UIViewController {
    
    let webView: CustomWKWebView
    let model: PrintArticle
    
    init(_ model: PrintArticle) {
        
        self.model   = model
        self.webView = CustomWKWebView(isNavigationGestrue: true)
        
        super.init(nibName: nil, bundle: nil)
        
        self.title = model.title
    }
    
    required init?(coder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    func setupUI() {
        
        self.view.addSubview(self.webView)
        
        self.webView.snp.makeConstraints({m in
            
            m.edges.equalToSuperview()
        })
        
        self.webView.requestUrl(requestUrl: self.model.newsUrl)
        
        self.webView.webviewDelegate = self
    }
    
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        self.setupUI()
        
        self.navigationController?.navigationBar.tintColor = .darkGray
    }
}

extension DetailNewsViewController: CustomWKWebViewDelegate {
    
    func isHttpRequestUrl(callUrl: URL?) -> Bool {
        
        return true
    }
    
    func loadingFinish() {
        
        guard (self.webView.title?.isEmpty ?? true) == false else {
            
            return
        }
        
        self.title = self.webView.title
    }
}
